#! /bin/sh /usr/share/dpatch/dpatch-run
## 27_check_radius_segfault.dpatch by Yasper Casper <komputnik@gmail.com>
##

@DPATCH@
diff -urNad nagios-plugins-1.4.5~/plugins/check_radius.c nagios-plugins-1.4.5/plugins/check_radius.c
--- nagios-plugins-1.4.5~/plugins/check_radius.c	2006-10-20 01:53:28.000000000 +0200
+++ nagios-plugins-1.4.5/plugins/check_radius.c	2006-11-20 20:38:49.000000000 +0100
@@ -137,6 +137,14 @@
 			rc_read_dictionary (rc_conf_str (str)))
 		die (STATE_UNKNOWN, _("Config file error"));
 
+	/* Initialize Value Pair to prevent segfault on rc_avpair_add. 
+	 * debugging on radiuslib-ng show that the mem isn't allocated,
+	 * and some random memory is in the USER-PASSWORD pair.
+	 * So, after initialization, the password is filled with the correct values 
+	 */
+	data.send_pairs = NULL;
+	data.receive_pairs = NULL;
+
 	service = PW_AUTHENTICATE_ONLY;
 
 	if (!(rc_avpair_add (&data.send_pairs, PW_SERVICE_TYPE, &service, 0) &&
